### PR TITLE
add aliases schema & check it

### PR DESCRIPTION
**Pull request by nevermore3**
refer to https://github.com/vesoft-inc/nebula-graph/issues/869

### CODE REVIEW

File: tests/tck/features/match/SeekByTag.intVid.feature [hunk 1]
```diff
@@ -18,9 +18,7 @@ Feature: Match seek by tag
       MATCH (v:bachelor)
       RETURN id(v) AS vid, v.age AS age
       """
-    Then the result should be, in any order, and the columns 0 should be hashed:
-      | vid          | age |
-      | 'Tim Duncan' | 42  |
+    Then a SemanticError should be raised at runtime: Not found prop `age'
```
Comment (Shylock-Hg) [inline tests/tck/features/match/SeekByTag.intVid.feature:21]: It's runable.

Comment (nevermore3) [reply tests/tck/features/match/SeekByTag.intVid.feature:21]: bachelor has no  age attribute， how could it be workable

Comment (Shylock-Hg) [reply tests/tck/features/match/SeekByTag.intVid.feature:21]: The vertex has it. I think you could check the properties in this space and report error if don't exists.

**Review by yixinglu — APPROVED** [review]

**Comment by czpmango** [comment]
Will the behavior of schema checking be consistent with neo4j or follow the cypher standard?

**Comment by czpmango** [comment]
IMO, this may conflict with the original intention of `cypher` language design.

**Comment by nevermore3** [comment]
> Will the behavior of schema checking be consistent with neo4j or follow the cypher standard?

We are a strong schema type, but cyhper is not, so I think we can refer to mysql's type checking mechanism

**Comment by czpmango** [comment]
> > Will the behavior of schema checking be consistent with neo4j or follow the cypher standard?
> 
> We are a strong schema type, but cyhper is not, so I think we can refer to mysql's type checking mechanism

`nGql` or `match`?

**Comment by nevermore3** [comment]
> > > Will the behavior of schema checking be consistent with neo4j or follow the cypher standard?
> > 
> > 
> > We are a strong schema type, but cyhper is not, so I think we can refer to mysql's type checking mechanism
> 
> `nGql` or `match`?

both

File: tests/tck/features/match/Base.IntVid.feature [hunk 2]
```diff
@@ -406,6 +405,38 @@ Feature: Basic match
       | [:like "Tony Parker"->"Manu Ginobili" @0 {likeness: 95}]                             |
       | [:like "Tony Parker"->"Tim Duncan" @0 {likeness: 95}]                                |
 
+  Scenario: check Alias attribute
```
Comment (czpmango) [inline tests/tck/features/match/Base.IntVid.feature:408]: Could you add test case like this:
```
MATCH (a:player)-[b]-> (c) where c.age>45 or c.name >= "Heat" or a.noexist>2.3 return c
```

**Comment by czpmango** [comment]
> > > > Will the behavior of schema checking be consistent with neo4j or follow the cypher standard?
> > > 
> > > 
> > > We are a strong schema type, but cyhper is not, so I think we can refer to mysql's type checking mechanism
> > 
> > 
> > `nGql` or `match`?
> 
> both

This is quite different from cypher. Do we still need to ensure that the query result is compatible with cypher? How do you deal with short-circuit logic?

File: tests/tck/features/match/Base.IntVid.feature (hunk 2 above)
Comment (nevermore3) [reply tests/tck/features/match/Base.IntVid.feature:408]: already exist in RelationalExpr.feature

File: tests/tck/features/match/SeekByTag.feature [hunk 3]
```diff
@@ -18,9 +18,7 @@ Feature: Match seek by tag
       MATCH (v:bachelor)
       RETURN id(v) AS vid, v.age AS age
       """
-    Then the result should be, in any order:
-      | vid          | age |
-      | 'Tim Duncan' | 42  |
+    Then a SemanticError should be raised at runtime: Not found prop `age'
```
Comment (yixinglu) [inline tests/tck/features/match/SeekByTag.feature:21]: `(v:bachelor)` doesn't mean vertex `v` only has one tag named `bachelor`. It means `v` must have the tag `bachelor` and maybe have multiple other tags.

Comment (yixinglu) [reply tests/tck/features/match/SeekByTag.feature:21]: So this limitation check is not a reasonable way.

**Pull request closed without merging**